scipy/stats/tests/test_morestats.py

```diff
@@ -2570,18 +2548,47 @@ def test_nan_propagate(self, test_func):
                                 1: 0.005545914017677123}),
                               (stats.circstd,
                                {None: np.nan, 0: 4.11093193, 1: 6.04265394})])
-    def test_nan_propagate_array(self, test_func, expected):
-        x = np.array([[355, 5, 2, 359, 10, 350, 1],
-                      [351, 7, 4, 352, 9, 349, np.nan],
-                      [1, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan]])
+    def test_nan_propagate_array(self, test_func, expected, xp):
+        x = xp.asarray([[355, 5, 2, 359, 10, 350, 1],
+                        [351, 7, 4, 352, 9, 349, np.nan],
+                        [1, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan]])
         for axis in expected.keys():
             out = test_func(x, high=360, axis=axis)
             if axis is None:
-                assert_(np.isnan(out))
+                xp_assert_equal(out, xp.asarray(xp.nan))
             else:
-                assert_allclose(out[0], expected[axis], rtol=1e-7)
-                assert_(np.isnan(out[1:]).all())
+                rtol = 2e-5 if is_torch(xp) else 1e-7
+                xp_assert_close(out[0], xp.asarray(expected[axis]), rtol=rtol)
```

Comment:
<details>
<summary> torch GPU failure observed locally </summary>

```
scipy/stats/tests/test_morestats.py:2561: in test_nan_propagate_array
    xp_assert_close(out[0], xp.asarray(expected[axis]), rtol=rtol)
E   AssertionError: Scalars are not close!
E   
E   Expected 0.2872505187988281 but got 0.28726011514663696.
E   Absolute difference: 9.59634780883789e-06 (up to 0 allowed)
E   Relative difference: 3.340759086864716e-05 (up to 2e-05 allowed)
```

</details>

`3.35e-5` for `torch` was sufficient on GPU locally

Reply:
Ok, another one best left to the new default.